### PR TITLE
Updates to Queue Atoms/Beans

### PR DESCRIPTION
**Pull request by mtwharmby**
This PR updates all of the queue atoms/beans with a shortname for use with a currently under development QueueBeanFactory. It also restructures the *Process classes to remove boilerplate code from the concrete classes. Finally it makes changes to MonitorAtomProcess that fix the unit test for termination. As part of this, I re-structured MonitorAtomProcess so it's more like my other *Process classes; I'm happy to make changes to improve readability if felt necessary.

### CODE REVIEW

File: org.eclipse.scanning.event.queues/src/org/eclipse/scanning/event/queues/processes/MonitorAtomProcess.java [hunk 1]
```diff
@@ -64,131 +69,134 @@
...
 
-			//TODO Additional abort action, not handled as part of run()?
-			terminated = true;
+	@Override
+	protected void postMatchAnalysis() throws EventException, InterruptedException {
+		if (isTerminated()) {
+			queueBean.setMessage("Get value of '"+queueBean.getMonitor()+"' aborted (requested)");
+			logger.debug("'"+bean.getName()+"' was requested to abort");
+			return;
+		} else if (queueBean.getPercentComplete() >= 99.5 && !terminated) {
```
Comment (gerring) [inline org.eclipse.scanning.event.queues/src/org/eclipse/scanning/event/queues/processes/MonitorAtomProcess.java]: In another place we have  "99.49) {//99.49 to catch rounding errors" Why not here?
Also those blocks seem to be if (terminaled) ... else if (>99.5) ... else ...  This logic should appear once ideally only the three actions ... being different.

File: org.eclipse.scanning.event.queues/src/org/eclipse/scanning/event/queues/processes/PositionerAtomProcess.java [hunk 2]
```diff
@@ -132,69 +135,44 @@ public synchronized void run() {
...
-
-			/*
-			 * N.B. Broadcasting needs to be done last; otherwise the next 
-			 * queue may start when we're not ready. Broadcasting should not 
-			 * happen if we've been terminated.
-			 */
-			if (!isTerminated()) {
-				broadcast();
-			}
+		if (isTerminated()) {
```
Comment (gerring) [inline org.eclipse.scanning.event.queues/src/org/eclipse/scanning/event/queues/processes/PositionerAtomProcess.java]: Same repeated logic again

File: org.eclipse.scanning.event.queues/src/org/eclipse/scanning/event/queues/processes/SubTaskAtomProcess.java [hunk 3]
```diff
@@ -54,82 +54,40 @@ public SubTaskAtomProcess(T bean, IPublisher<T> publisher, Boolean blocking) thr
...
-			processLatch.countDown();
-			
-			//Wait for post-match analysis to finish
-			continueIfExecutionEnded();
-		} catch (InterruptedException iEx) {
-			throw new EventException(iEx);
-		} finally {
-			postMatchAnalysisLock.unlock();
+		if (isTerminated()) {
+			atomQueueProcessor.terminate();
```
Comment (gerring) [inline org.eclipse.scanning.event.queues/src/org/eclipse/scanning/event/queues/processes/SubTaskAtomProcess.java]: I've seen this somewhere before. Is it a remake?

File: org.eclipse.scanning.event.queues/src/org/eclipse/scanning/event/queues/processes/TaskBeanProcess.java [hunk 4]
```diff
@@ -57,85 +57,43 @@ public TaskBeanProcess(T bean, IPublisher<T> publisher, Boolean blocking) throws
...
-			logger.debug("Termination of '"+queueBean.getName()+"' requested; release processLatch (start post-match analysis)");
-			processLatch.countDown();
-			
-			//Wait for post-match analysis to finish
-			continueIfExecutionEnded();
-		} catch (InterruptedException iEx) {
-			throw new EventException(iEx);
-		} finally {
-			postMatchAnalysisLock.unlock();
+		if (isTerminated()) {
```
Comment (gerring) [inline org.eclipse.scanning.event.queues/src/org/eclipse/scanning/event/queues/processes/TaskBeanProcess.java]: Hello again old friend

**Comment by mtwharmby** [comment]
Thanks for your comments @gerring I'll have a look over it and rework as necessary.

File: org.eclipse.scanning.test/src/org/eclipse/scanning/test/event/queues/dummy/DummyBeanProcess.java [hunk 5]
```diff
@@ -27,5 +27,20 @@ public DummyBeanProcess(T bean, IPublisher<T> publisher, Boolean blocking) throw
 	public Class<DummyBean> getBeanClass() {
 		return DummyBean.class;
 	}
+
```
Comment (gerring) [inline org.eclipse.scanning.test/src/org/eclipse/scanning/test/event/queues/dummy/DummyBeanProcess.java]: You could put this once in the original interface:

 +	default void postMatchCompleted() throws EventException {
 +		//Implement as required
 +	}
 +
 +	default void postMatchTerminated() throws EventException {
 +		//Implement as required
 +	}
 +
 +	default void postMatchFailed() throws EventException {
 +		//Implement as required
 +	}

Comment (mtwharmby) [inline org.eclipse.scanning.test/src/org/eclipse/scanning/test/event/queues/dummy/DummyBeanProcess.java]: Done.

**Comment by mtwharmby** [comment]
@gerring Just pushed up your suggested changes and I've tested it with the server running in the IDE - the PositionerAtom seems to be consumed fine, which it wasn't being before. So...

@Mark-Booth Once the tests pass, can you build a new version of GDA which we can test on the beamline based on these changes please? Thanks!

**Comment by mtwharmby** [comment]
Test ActivemqConnectorServiceJsonMarshallingTest.testObjectArrayDeserialization:176 » IllegalArgument is failing. I haven't made any changes to it and nor can I reproduce the failure on my own machine. The test is that simple that I suspect something might be amiss with libraries?

 @gerring  I will mark it ignored in the next commit.

**Comment by Mark-Booth** [comment]
Are we happy to merge this in now @gerring ?

**Comment by mtwharmby** [comment]
I have reverted my change to ignore the testObjectArrayDeserialization. I don't think marking a test ignored in a key component like the activemq connector is sensible, though I don't see what the problem with the failed test is/was.

**Comment by mtwharmby** [comment]
Reverted the revert, since @gerring 's change didn't go in, due to one of my tests failing. A fix for that failing test (RemoteQueueControllerServiceTest) is in this PR.

**Comment by mtwharmby** [comment]
Success! Finally! @gerring is this OK then?

**Comment by gerring** [comment]
Let's put this back and remove the @Ignore from my change and see what happens...

**Pull request merged**